src/PhotoPage.vala

```diff
@@ -654,16 +663,49 @@ public abstract class EditingHostPage : SinglePhotoPage {
         return true;
     }
 
-    private bool on_zoom_out_pressed (Gdk.EventButton event) {
-        snap_zoom_to_min ();
+    private bool zoom_callback () {
+        if (active_zoom_widget == zoom_in_box) {
+            activate_action ("IncreaseSize");
+        } else {
+            activate_action ("DecreaseSize");
+        }
+
+        return true;
+    }
+
+    private bool on_zoom_button_pressed (Gdk.EventButton event, Gtk.EventBox sender) {
+        clear_zoom_timeouts ();
+
+        active_zoom_widget = sender;
+        zoom_initial_wait_id = Timeout.add (settings.gtk_timeout_initial, () => {
+            zoom_click_id = Timeout.add (settings.gtk_timeout_repeat, zoom_callback);
+            zoom_initial_wait_id = 0;
+            return false;
+        });
+
+        zoom_callback ();
+
         return true;
     }
 
-    private bool on_zoom_in_pressed (Gdk.EventButton event) {
-        snap_zoom_to_max ();
+    private bool on_zoom_button_released (Gdk.EventButton event, Gtk.EventBox sender) {
+        clear_zoom_timeouts ();
+
         return true;
     }
 
+    private void clear_zoom_timeouts () {
```

Comment:
Should really call this on leave-notify event in case button has not been released. (Perhaps unlikely but possible)